AppConnectSwift/CaptureImageViewController.swift

```diff
@@ -11,26 +11,68 @@ import UIKit
 class CaptureImageViewController: UIViewController, UIImagePickerControllerDelegate, UINavigationControllerDelegate {
     
     @IBOutlet weak var imageView: UIImageView!
+    @IBOutlet weak var saveImage: UIBarButtonItem!
+    
     var imagePicker = UIImagePickerController()
     var image = UIImage()
+    var userID : Int64!
+    var data : NSData!
+    var collectedSubjects: [MDSubject]!
+    var datastore = MDDatastoreFactory.create()
     
     override func viewDidLoad() {
         super.viewDidLoad()
         takeOrSelectPicture(true)
         imagePicker.allowsEditing = false
         imagePicker.delegate = self
+        self.saveImage.enabled = false
+        self.loadSubjects()
     }
     
     override func viewDidAppear(animated: Bool) {
         super.viewDidAppear(animated)
         self.navigationItem.title = "Capture Image"
     }
     
+    func loadSubjects() {
+        var bgQueue : NSOperationQueue! = NSOperationQueue()
+        bgQueue.addOperationWithBlock() {
+            let clientFactory = MDClientFactory.sharedInstance()
+            let client = clientFactory.clientOfType(MDClientType.Network);
+            let user = self.datastore.userWithID(Int64(self.userID))
```

Comment:
There may be some issues with using a datastore created on a different thread. I think elsewhere in the app we create a new datastore within the background block.